src/layers/domain/response/render_response.py

```diff
@@ -33,19 +36,31 @@ def render_response[
     if isinstance(response, Exception):
         response = validate_exception(response)
 
-    if isinstance(response, HTTPStatus):
+    if response == HTTPStatus.NO_CONTENT:
+        http_status = response
+        outcome = None
+    elif isinstance(response, HTTPStatus):
         # Explicit success (e.g. CREATE, DELETE, UPDATE operations)
         http_status = response
         outcome = operation_outcome_ok(id=id, http_status=http_status)
+    elif isinstance(response, ValidationError):
+        # Implicit failure from ValidationError
+        outcome = ErrorResponse.from_validation_error(exception=response).dict()
+        http_status = http_status_from_exception(exception=response)
     elif isinstance(response, Exception):
-        # Implicit failure
-        http_status, outcome = operation_outcome_not_ok(id=id, exception=response)
+        # Implicit failure from all other Exceptions
+        outcome = ErrorResponse.from_exception(exception=response).dict()
+        http_status = http_status_from_exception(exception=response)
     else:
-        # Implicit success (e.g. SEARCH, READ operations)
-        http_status = HTTPStatus.OK
-        outcome = response
+        if isinstance(response, tuple):
+            http_status = response[0]
+            outcome = response[1]
```

Comment:
To be explicit (meaning we're also requiring there to be exactly do items in the tuple)

```suggestion
            http_status, outcome = response
```